### PR TITLE
Gotype/fix

### PR DESCRIPTION
**Pull request by gandhikrishna**
Currently YARPC proto plugin levrages `{message}.GetName()` to serve as the GoType. This is the value directly coming from proto IDL , so `Get2ndMessageRequest` remains as `Get2ndMessageRequest`

But `gogo/protobuf` uses CamelCase to serve as the GoType. Hence `Get2ndMessageRequest`(from IDL) becomes `Get2NdMessageRequest` in protobuf files(*.pb.go). Since there is a imparity between gogoproto and yarpc message name the build starts failing with error like 
`example.pb.yarpc.go: undefined: Get2ndMessageRequest`

The aim of this diff is to create parity between YARPC proto plugin GoType and gogo/protobuf proto plugin GoType by using gogo's CamelCase API.

- [x] Description and context for reviewers: one partner, one stranger
- [x]  Entry in CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - peer/direct: peer connections were closed even if they were still in use.
 - configuration: clarify error message for the special case of attempting to use a peer list
   updater when none have been registered.
+- protoplugin: Yarpc plugin GoType conventions aligned with gogo/protobuf.
```
Comment (AllenLuUber) [inline CHANGELOG.md:12]: ~~Yarpc plugin~~ since we are already in the protoplugin context

File: internal/protoplugin/protoplugin.go [hunk 2]
```diff
@@ -199,7 +200,7 @@ func (m *Message) FQMN() string {
 func (m *Message) GoType(currentPackage string) string {
 	var components []string
 	components = append(components, m.Outers...)
-	components = append(components, m.GetName())
+	components = append(components, gogogen.CamelCase(m.GetName()))
```
Comment (AllenLuUber) [inline internal/protoplugin/protoplugin.go:203]: can you add comment to link to the line where CamelSlice is called in gogoproto?

**Review by AllenLuUber — APPROVED** [review]

File: internal/protoplugin/protoplugin.go (hunk 2 above)
Comment (gandhikrishna) [reply internal/protoplugin/protoplugin.go:207]: Done

File: internal/prototest/examplepb/example.proto [hunk 3]
```diff
@@ -45,6 +45,14 @@ message EchoBothResponse {
...
 }
 
+message Get2ndMessageRequest {
+   string message = 1;
+}
+
+message Get2ndMessageResponse {
+  string message = 1;
+}
+
```
Comment (Dogild) [inline internal/prototest/examplepb/example.proto:55]: Please add a comment on the message on why we decided to add this type of message name.

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - peer/direct: peer connections were closed even if they were still in use.
 - configuration: clarify error message for the special case of attempting to use a peer list
   updater when none have been registered.
+- protoplugin: GoType conventions aligned with gogo/protobuf.
```
Comment (Dogild) [inline CHANGELOG.md:12]: I am not very convinced about this changelog entry. Should it be more:
- protoplugin: generated code follows go type conventions from gogo/protobuf.

**Pull request merged**